inc/media.php

```diff
@@ -2422,6 +2423,27 @@ function media_resize_imageGD($ext,$from,$from_w,$from_h,$to,$to_w,$to_h,$ofs_x=
         imagecopyresized($newimg, $image, 0, 0, $ofs_x, $ofs_y, $to_w, $to_h, $from_w, $from_h);
     }
 
+    // clean the old image up before we try to rotate the new image.
+    if($image) imagedestroy($image);
+
+    // rotate the image based on the EXIF data.
+    if(function_exists("exif_read_data")){
```

Comment:
This should use our JPEGMeta class instead.

Reply:
Just pushed that change. Definitely looks much nicer.